### PR TITLE
added status on separated thred to show every 30 sec

### PR DESCRIPTION
**Pull request by Ionel14**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -4,36 +4,63 @@
...
+            // devicesThread.detach();
+
+            smart_home::SmartHomeStatus status;
+            
+            std::thread statusThread([&smartHome, &status]() {
+                    status.PrintSmartHomeStatus(smartHome);
+
+                });
+
+            statusThread.detach();
```
Comment (CristianIonutIonescu) [inline main.cpp]: try to use the threads with join, not detach as you share data between them.
Also there is no synchronisation need between the threads (even though you sleep the main thread and do changes). Look up lock_guard, unique_lock, read about mutexes and how you should include them in your project to make use of them

**Review by CristianIonutIonescu — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 1 above)
Comment (Ionel14) [reply main.cpp]: I tried to use that thread with join, but if I do that my main thread sleeps, and I understood from my homework that I should make changes without any issues, so how am I supposed to do that if my main thread sleeps when I do join?
I used unique_lock and mutex in the SmartHomeStatus class, and I understood that they are used for avoiding data races, but in this case I don't know how they can be useful

Comment (Ionel14) [reply main.cpp]: I used another thread in my last commit to do changes, and I used all threads with join.

**Review by CristianIonutIonescu — APPROVED** [review]
